### PR TITLE
Write a systemd service file when using RedHat >= 7

### PR DESCRIPTION
**Pull request by Flupsy**
The module wasn't able to restart on a RHEL 7 box because a non-default interface name was being used and the systemd service file wasn't being updated.

### CODE REVIEW

File: spec/classes/dhcp_spec.rb [hunk 1]
```diff
@@ -22,6 +22,7 @@
         osfamily: 'RedHat',
         operatingsystem: 'RedHat',
         operatingsystemrelease: '6',
+        operatingsystemmajrelease: '6',
```
Comment (petems) [inline spec/classes/dhcp_spec.rb]: Could you add a test for when the major release is 7, adding in the detection of the service file? 👍

Comment (Flupsy) [inline spec/classes/dhcp_spec.rb]: I can do, but there's no test at the moment for `/etc/sysconfig/dhcpd` either.  Should I add one?

**Comment by dhoppe** [comment]
I know this is an RedHat specific PR, but I think Debian 8 and Ubuntu 16.04 will require this too.

**Comment by jyaworski** [comment]
I believe there's a fact called $service_provider that's better suited to determine this use.

**Comment by Flupsy** [comment]
I don't see `service_provider` in facter output from RHEL 6 or 7 systems.

**Comment by ekohl** [comment]
> I don't see service_provider in facter output from RHEL 6 or 7 systems (facter 3.4.1, puppet 4.7.0).

That's because puppetlabs-stdlib provides the fact (since 4.10.0)

@Flupsy how about using a drop in file where you just override the command?

**Comment by ekohl** [comment]
For reference: https://github.com/theforeman/puppet-dhcp/pull/110 is how I implemented it in theforeman/dhcp.

**Comment by petems** [comment]
This has been floating around for a while, @Flupsy, you mind if I rebase this and see if we can get it merged up? 😄

**Comment by Flupsy** [comment]
@petems Yes please! :)

**Comment by petems** [comment]
Closing in favour of https://github.com/voxpupuli/puppet-dhcp/pull/154

**Pull request closed without merging**